### DISCUSSION TITLE
How do you handle windows being on the wrong monitor after wake from sleep?

### DISCUSSION THREAD

**Discussion by mo-st** [asmvik/yabai#2258] [Q&A]
After wake from sleep, sometimes all the windows carefully placed across my three monitors will be smushed together on the main one, the others being empty. I've been having that problem for years with Amethyst and assumed that it was problem of that app, but after a bit of searching it seems that this is a [general problem](https://discussions.apple.com/thread/253803495) of MacOS. 

The only workaround I am aware of is disabling sleep when plugged in altogether (as shown in the apple forum post above), at least I can continue working after a short break right away, ~this only works if I don't close the lid of the laptop however~. [Edit: it seems to circumvent the issue from occuring and so far yabai was able to correctly place all windows even after closed lid, sleep and unpluging monitors. Disabling sleep isn't ideal but it works at least] Have any of you figured out another solution to this problem? Is yabai capable of solving this reliably?

Thanks for sharing your experience with this issue

**Comment by net** [comment]
I don't have a solution, but I'd also love to know if anyone else does!

The problem I usually experience is windows being in the wrong space (as in macOS space) after removing or adding monitors, which I do a lot when switching between working at a desk and working on just my laptop. They also seem to loose their yabai configuration, although that may be a side effect of them being in the wrong space.

I wonder if yabai could track space attributes for each window, and re-apply them on wake and on display configuration change (and relayout the windows across all spaces, in case space dimensions change too).

**Reply by mo-st** [reply]
The solution with disabling sleep has been working pretty great for me so far. Since then I never had any issue with wrong screens or spaces, even when switching around monitors. I think with sleep disabled yabai actually is able to track exactly what you suggested, at least it seems that way to me. Starting up with one monitor disconnected in the morning results in a mess, but as soon as I plug it in, all windows are thrown back into the appropriate place.

**Comment by qns7** [comment]
It's not perfect, but memmon works quite fine for this: https://github.com/relikd/Memmon
